registry/handlers/referrers.go

```diff
@@ -80,6 +97,19 @@ func (h *referrersHandler) GetReferrers(w http.ResponseWriter, r *http.Request)
 		referrers = []v1.Descriptor{}
 	}
 
+	// only consider pagination if # of referrers is greater than page size
+	if len(referrers) > pageSize {
+		startIndex := pageNumber * pageSize
+
+		// if there's only 1 page of results left
+		if len(referrers)-startIndex <= pageSize {
+			referrers = referrers[startIndex:]
```

Comment:
What if `startIndex` is larger than `len(referrers)`? It will panic.

Reply:
Didn't think about this case. Added relevant handling: if `startIndex` is out of bound returns `referrers = []v1.Descriptor{}`.